bin/detect_reads.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+import argparse
+import csv
+import sys
+from typing import List, NoReturn
+
+
+def parse_args(args=None) -> argparse.Namespace:
+    """
+    Reformatting is based on detecting whether the reads are paired or single end.
+    Script appends appropriate column to samplesheet.csv file.
+    """
+    Description = "Reformat nf-core/taxprofiler samplesheet file."
+    Epilog = "Example usage: python detect_reads.py <FILE_IN> <FILE_OUT>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument("FILE_IN", help="Input samplesheet file.")
+    parser.add_argument("FILE_OUT", help="Output file.")
+    return parser.parse_args(args)
+
+
+class ReadsModifier:
+    def __init__(self):
+        self.headers = None
+        self.sample_index = None
+        self.fastq_1_index = None
+        self.fastq_2_index = None
+        self.fasta_index = None
+
+    def detect_reads_and_reformat(self, input_file_path: str, output_file_path: str) -> NoReturn:
+        NEW_COLUMN_NAME = "single_end"
+        new_file_rows = []
+
+        with open(input_file_path, "r") as input_file:
```

Comment:
Best practice is to open the file with `newline=""` (https://docs.python.org/3/library/csv.html#id3):

```suggestion
        with open(input_file_path, "r", newline="") as input_file:
```